app/models/api/messages/flowcell_io.rb

```diff
@@ -96,8 +96,9 @@ def detect_float_descriptor(name, ignored_unit)
         end
 
         def detect_descriptor(name)
-          lab_events.each { |e| e.descriptor_value_for(name).tap { |bc| return bc if bc.present? } }
-          nil # We have no flowcell barcode
+          # Sort here goes by id. Which should cover our needs
```

Comment:
```suggestion
          # Sort here goes by id descending, in order to find the most recent descriptor with the passed in 'name'
```

What does variable `bc` stand for?

Reply:
Sort actuall is ascending, which is why we do a reverse_each

Reply:
But as to bc, that's a very good question. I ket the previous variable names in place, but that could definitely do with a change.

I did attempt a large change here, but couldn't work anything else out.

Reply:
Huh, although no reason to tap this, as we're not mapping.